### PR TITLE
feat: support for create/drop/desc/show connection.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

start of https://github.com/datafuselabs/databend/issues/13489
need another pr to use the created connection

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13627)

### CODE REVIEW

File: src/meta/protos/proto/connection.proto [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+package databend_proto;
+
+message UserDefinedConnection {
+  uint64 ver = 100;
+  uint64 min_reader_ver = 101;
+
+  string name = 1;
+  string storage_type = 2;
+  map<string, string> storage_params= 3;
```
Comment (Xuanwo) [inline src/meta/protos/proto/connection.proto:25]: Is map in proto3 ordered?

File: src/query/ast/tests/it/testdata/statement.txt [hunk 2]
```diff
@@ -13593,6 +13593,74 @@ DescribeTask(
 )
 
 
+---------- Input ----------
+CREATE CONNECTION IF NOT EXISTS my_conn STORAGE_TYPE='s3'
```
Comment (Xuanwo) [inline src/query/ast/tests/it/testdata/statement.txt:13597]: Would you like to add a test case for CREATE CONNECTION with args like root and endpoint?

File: src/query/sql/src/planner/binder/ddl/connection.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        stmt: &CreateConnectionStmt,
+    ) -> Result<Plan> {
+        let mut location = UriLocation::new(
+            stmt.storage_type.clone(),
+            "".to_string(),
+            "".to_string(),
+            "".to_string(),
+            stmt.storage_params.clone(),
+        );
+        parse_uri_location(&mut location).await?;
```
Comment (Xuanwo) [inline src/query/sql/src/planner/binder/ddl/connection.rs:37]: We cab perform a `StorageParams::auto_detect()` here to make sure it's valid.

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 10 of 47 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13627)* status: 10 of 47 files reviewed, 3 unresolved discussions (waiting on @andylokandy, @Xuanwo, and @youngsofun)

___
*[`src/meta/protos/proto/connection.proto` line 25 at r1](https://reviewable.io/reviews/datafuselabs/databend/13627#gh-1385854098:-NighAYm49-4SEv_g23P:bttu1pn) ([raw file](https://github.com/datafuselabs/databend/blob/586cd9fc36ea8dac6a589c0a472dde9e83e5577f/src/meta/protos/proto/connection.proto#L25)):*
<details><summary><i>Previously, Xuanwo (Xuanwo) wrote…</i></summary><blockquote>

Is map in proto3 ordered?
</blockquote></details>

Yes 
https://github.com/datafuselabs/fuse-query/blob/a6df2f6949ec7a66e3002ee6420457ec71cb0ff7/src/meta/protos/build.rs#L72

File: src/query/ast/tests/it/testdata/statement.txt (hunk 2 above)
Comment (youngsofun) [reply src/query/ast/tests/it/testdata/statement.txt:13597]: root is not valid arg for now?
do you mean a valid arg and a invalid or 2 valid args?

File: src/query/sql/src/planner/binder/ddl/connection.rs (hunk 3 above)
Comment (youngsofun) [reply src/query/sql/src/planner/binder/ddl/connection.rs:37]: already done inside parse_uri_location

File: src/meta/protos/proto/connection.proto (hunk 1 above)
Comment (youngsofun) [reply src/meta/protos/proto/connection.proto:25]: you are right, thanks!

Comment (youngsofun) [reply src/meta/protos/proto/connection.proto:25]: > Is map in proto3 ordered?

not ordered. but we do not rely on this order.

File: src/query/ast/tests/it/testdata/statement.txt (hunk 2 above)
Comment (youngsofun) [reply src/query/ast/tests/it/testdata/statement.txt:13597]: done

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**